plugins/optimization-detective/tests/test-optimization.php

```diff
@@ -291,6 +291,18 @@ public function data_provider_test_od_can_optimize_response(): array {
 				},
 				'expected' => false,
 			),
+			'singular_as_post_preview'             => array(
+				'set_up'   => static function (): string {
+					$post_id = self::factory()->post->create(
+						array(
+							'post_title'  => 'Hello',
+							'post_status' => 'draft',
+						)
+					);
+					return get_preview_post_link( $post_id );
```

Comment:
Since we know it will actually return a string:

```suggestion
					return (string) get_preview_post_link( $post_id );
```

Reply:
Oh yes, we can explicitly typecast it 👍.